File: ruby/6.rb
# project euler 6: find (1 + 2 + ... + 100)^2 - (1^2 + 2^2 + ... + 100^2)
t = Time.new.to_f
sum1,sum2 = 0,0
(1..100).to_a.each {|n| sum1 += n}
(1..100).to_a.each {|n| sum2 += n**2}
result = sum1**2 - sum2
puts "result: #{result} - found in #{((Time.new.to_f - t)*1000).round 3} ms"

File: c/6.c
#include <stdio.h>

/* Project Euler 6
 * Find the difference between the square of the sum of 1-100 (1...100)^2 and the sum of the squares of 1-100 (1^2...100^2)
 */
int main(int argc, char *argv[])
{
    int square_sum = 0;
    int sum_of_squares = 0;
    int difference;

    int i;
    for (i = 1; i <= 100; i++) {
        square_sum += i;
        sum_of_squares += (i * i);
    }

    difference = (square_sum * square_sum) - sum_of_squares;
    printf("Result: %d\n", difference);

    return 1;
}
